### PR TITLE
add token to request to private orbs

### PR DESCRIPTION
**Pull request by abdelDriowya**

### CODE REVIEW

File: pkg/server/methods/executeCommand.go [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+func (methods *Methods) ExecuteCommand(reply jsonrpc2.Replier, req jsonrpc2.Request) error {
+	params := protocol.ExecuteCommandParams{}
+	if err := json.Unmarshal(req.Params(), &params); err != nil {
+		return reply(methods.Ctx, nil, fmt.Errorf("%s: %w", jsonrpc2.ErrParse, err))
+	}
+
+	token := params.Arguments[0].(string)
+	methods.Cache.TokenCache.SetToken(token)
+	return reply(methods.Ctx, nil, nil)
```
Comment (rlegan) [inline pkg/server/methods/executeCommand.go:19]: we should double check if the command is `setToken`, just in case ! 😄

File: pkg/utils/cache.go [hunk 2]
```diff
@@ -11,6 +11,12 @@ type Cache struct {
 	FileCache   FileCache
 	OrbCache    OrbCache
 	DockerCache DockerCache
+	TokenCache  TokenCache
+}
+
+type TokenCache struct {
+	cacheMutex *sync.Mutex
+	Token      string
```
Comment (rlegan) [inline pkg/utils/cache.go:19]: change `Token` to `token` to make it and private, use the mutex and the function `SetToken`

**Comment by rlegan** [comment]
note: use `webUI: cci-internal/web-ui-orb@1.0.0` orb to test if whether it works or not

**Comment by rlegan** [comment]
when removing the token, the private orbs are no longer unvalid, which should be the case
the issue is the cache not being updated, we still have the private orb in cache and the source of the orb downloaded: we should delete the source file of the private orbs and delete them from the cache

**Comment by sguilliaci** [comment]
Don't forget to base PR on `main` not `master` until ask-it changes the default branch

**Review by rlegan — APPROVED** [review]
lgtm

**Pull request merged**